Cabal/Distribution/PackageDescription/FieldGrammar.hs

```diff
@@ -432,17 +432,17 @@ optionsFieldGrammar
 optionsFieldGrammar = combine
     <$> monoidalFieldAla "ghc-options"   (alaList' NoCommaFSep Token') (extract GHC)
     <*> monoidalFieldAla "ghcjs-options" (alaList' NoCommaFSep Token') (extract GHCJS)
-    <*> monoidalFieldAla "jhc-options"   (alaList' NoCommaFSep Token') (extract JHC)
     -- NOTE: Hugs and NHC are not supported anymore, but these fields are kept
```

Comment:
We should either remove `{hugs,nhc98}-options` from here as well, or keep `jhc-options`.

Reply:
Removing those fields from the grammar would result in parser errors/warnings being emitted by newer lib:Cabal; if we want to remove them, we should only make it a parser warning/failure for newer `cabal-version` specs; while older specs should be silent about the occurence.

/cc @phadej

Reply:
Yeah, I've "moved" this to "knownField ..." for precisely the reasons that @hvr mentions.